internal/jobserver/server.go

```diff
@@ -0,0 +1,314 @@
+		Host:       loopback,
+		Port:       port,
+		DontListen: opts.NoListener,
+		Accounts:   []*server.Account{userAccount, systemAccount},
+		Users: []*server.User{
+			{Username: client.USERNAME, Password: client.NO_PASSWORD, Account: userAccount},
+			{Username: serverUsername, Password: noPassword, Account: userAccount},
+			{Username: sysUser, Password: noPassword, Account: systemAccount},
+		},
+		SystemAccount: systemAccount.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("creating NATS server instance: %w", err)
+	}
+
+	if opts.EnableLogging {
+		server.ConfigureLogger()
+	}
+
+	// Starting the server, and waiting for it to be ready
+	server.Start()
+
+	defer func() {
+		if srv == nil && err != nil {
+			// Shut down the server immediately, as we are returning an error and no
+			// server, so the caller wouldn't be able to do this by themselves.
+			server.Shutdown()
+		}
+	}()
+
+	if !server.ReadyForConnections(startTimeout) {
+		return nil, errors.New("timed out waiting for NATS server to become available")
+	}
+
+	var clientUrl string
+	if opts.NoListener {
+		// "Any" URL will do here, it's not actually used...
+		clientUrl = "nats://localhost:0"
+	} else {
+		// We don't use `server.ClientURL()` here because it currently returns an invalid URL is the
+		// listener address is IPv6 (see: https://github.com/nats-io/nats-server/issues/5721)
+		clientUrl = fmt.Sprintf("nats://%s", server.Addr())
+	}
+
+	log.Tracef("[JOBSERVER] NATS Server ready for connections on %q\n", clientUrl)
+
+	// Obtaining the local server connection
+	conn, err := nats.Connect(clientUrl, nats.UserInfo(serverUsername, noPassword), nats.InProcessServer(server))
+	if err != nil {
+		return nil, fmt.Errorf("connecting to in-process NATS server instance: %w", err)
+	}
+
+	// Installing the handlers
+	res := Server{
+		server:     server,
+		CacheStats: &common.CacheStats{},
+		clientUrl:  clientUrl,
+	}
+	if err := buildid.Subscribe(conn, res.CacheStats); err != nil {
+		return nil, err
+	}
+	if err := pkgs.Subscribe(clientUrl, conn, res.CacheStats); err != nil {
+		return nil, err
+	}
+	if _, err := conn.Subscribe("clients", res.handleClients); err != nil {
+		return nil, err
+	}
+
+	// Wait until all subscriptions have been processed by the server...
+	if err := conn.Flush(); err != nil {
+		return nil, err
+	}
+
+	if opts.InactivityTimeout > 0 {
+		sysConn, err := nats.Connect(clientUrl, nats.Name("server-local-admin"), nats.UserInfo(sysUser, noPassword), nats.InProcessServer(server))
+		if err != nil {
+			return nil, err
+		}
+
+		res.inactivityTimeout = opts.InactivityTimeout
+		res.clients = make(map[uint64]string)
+		if _, err := sysConn.Subscribe(fmt.Sprintf("$SYS.ACCOUNT.%s.CONNECT", userAccount.Name), res.handleClientConnect); err != nil {
+			return nil, err
+		}
+		if _, err := sysConn.Subscribe(fmt.Sprintf("$SYS.ACCOUNT.%s.DISCONNECT", userAccount.Name), res.handleClientDisconnect); err != nil {
+			return nil, err
+		}
```

Comment:
Are we sure this won't cause client-side issue if the handlers are not even subscribed?